jaxrs-api/src/main/java/jakarta/ws/rs/ext/FactoryFinder.java

```diff
@@ -123,15 +124,14 @@ static <T> Object find(final String factoryId, final Class<T> service) throws Cl
         }
 
         // try to read from $java.home/lib/jaxrs.properties
-        FileInputStream inputStream = null;
-        String configFile = null;
+        InputStream inputStream = null;
+        Path configFile = null;
         try {
             String javah = System.getProperty("java.home");
-            configFile = javah + File.separator + "lib" + File.separator + "jaxrs.properties";
-            File f = new File(configFile);
-            if (f.exists()) {
+            configFile = Path.of(javah, "lib", "jaxrs.properties");
+            if (Files.exists(configFile)) {
                 Properties props = new Properties();
-                inputStream = new FileInputStream(f);
+                inputStream = Files.newInputStream(configFile);
```

Comment:
I realize this was like this before, but this stream is never closed.

Reply:
This is not correct. The stream is closed here: https://github.com/jakartaee/rest/blob/ebced83761f9ffa59bdfc34c6ad2c4fddd71bbe7/jaxrs-api/src/main/java/jakarta/ws/rs/ext/FactoryFinder.java#L145.